File: src/dynamic-server/finger-memory-web/src/main/java/work/nobility/fingermemoryweb/common/ApiRestResponse.java
package work.nobility.fingermemoryweb.common;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import work.nobility.fingermemoryweb.exception.ExceptionEnum;

@AllArgsConstructor
@Getter
@Setter
@ToString
public class ApiRestResponse<T> {
  private Integer status;
  private String massage;
  private T data;

  public static final int OK_CODE = 10000;
  public static final String OK_MASSAGE = "SUCCESS";

  public ApiRestResponse(Integer status, String massage) {
    this.status = status;
    this.massage = massage;
  }

  public ApiRestResponse() {
    this(OK_CODE, OK_MASSAGE);
  }

  public static <T> ApiRestResponse<T> success() {
    return new ApiRestResponse<>();
  }

  public static <T> ApiRestResponse<T> success(T data) {
    ApiRestResponse<T> apiRestResponse = new ApiRestResponse<>();
    apiRestResponse.setData(data);
    return apiRestResponse;
  }

  public static <T> ApiRestResponse<T> error(Integer code, String massage) {
    return new ApiRestResponse<>(code, massage);
  }

  public static <T> ApiRestResponse<T> error(ExceptionEnum exceptionEnum) {
    return new ApiRestResponse<>(exceptionEnum.getCode(), exceptionEnum.getMessage());
  }
}

File: src/static-server/finger-memory-web/common/apiRestResponse.js
/**
 *  @swagger
 *  definitions:
 *    ApiRestResponse:
 *      properties:
 *        status:
 *          type: number
 *        massage:
 *          type: string
 *        data:
 *          type: object
 */
const OK_CODE = 10000;
const OK_MASSAGE = "SUCCESS";
class ApiRestResponse {
    constructor(status, massage, data) {
        this.status = status;
        this.massage = massage;
        this.data = data;
    }
}
ApiRestResponse.success = function success(data, status = OK_CODE,massage = OK_MASSAGE) {
    return new ApiRestResponse(status, massage, data);
}
ApiRestResponse.error = function error({status, massage}) {
    return new ApiRestResponse(status, massage, null);
}
module.exports = ApiRestResponse

